### PR TITLE
Add chex.exe par times for chex.wad.

### PR DESCRIPTION
**Pull request by Acts19quiz**
Potential changes to address issue #1355 but in short this is one (potential) way to add a new array containing the chex.exe par times. Default behaviour for chex.exe, if players happen to mess about and -warp or something into Ultimate DOOM levels, is for the DOOM music, pars, etc. to display, so the other 22 pars from Ultimate DOOM/"retail" are kept intact.

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
LGTM.

You could get away with a shorter array and less duplicate entries if you extended the check to e.g. `if (gameversion == exe_chex && gameepisode == 1)`.

Also, why `qpars[]`? Wouldn't e.g. `chexpars[]` sound better somehow?

File: src/doom/g_game.c
Comment (turol) [inline src/doom/g_game.c:1324]: This should be `static const`. So should `pars` and `cpars` above but that should be a separate commit (and PR). Can dehacked change these? Based on a quick source inspection it would seem not in the current implementation.

File: src/doom/g_game.c
Comment (turol) [inline src/doom/g_game.c:1326]: This should list all ten zeroes or GCC might warn about it being incomplete.

**Comment by Acts19quiz** [comment]
> You could get away with a shorter array and less duplicate entries if you extended the check to e.g. `if (gameversion == exe_chex && gameepisode == 1)`.
Duly noted. That's why I said "one potential way". :)
> Also, why `qpars[]`? Wouldn't e.g. `chexpars[]` sound better somehow?
It would definitely be better. I was just keeping form with the (super) short array names ("pars" for generic, "cpars" for commercial/DOOM II).
> This should be `static const`. So should `pars` and `cpars` above but that should be a separate commit (and PR). Can dehacked change these? Based on a quick source inspection it would seem not in the current implementation.
Everything that can be done with DeHackEd/chex.deh has been done. Par time changing is exclusive to BEX/Boom extensions, which I think is outside the scope of Chocolate Doom.

File: src/doom/g_game.c
Comment (fabiangreffrath) [reply src/doom/g_game.c:1326]: It won't. If you initialize one element of an array to whatever value, all the following ones are implicitly initialized to 0.

**Comment by fabiangreffrath** [comment]
@Acts19quiz Are you going to work on the change requests?

**Comment by Acts19quiz** [comment]
I had just assumed you'd either accept the pull request (an improbability since this is my first here) or if you liked the idea or concept of it, revise my revisions and use them in Choco proper (as it's a pull request for an open GPL project, I would of course be okay with that as per that little message the board here gave me when I filled out the form). Would you like for me to revise my PR here? I wouldn't know how to do that beyond making a new PR, though I could certainly give that a whack if that's what you want. Since I'm not familiar with the etiquette or procedures here, I wanted to defer to the maintainers on how they would have this handled.

**Comment by fabiangreffrath** [comment]
Choco being an open source project means that everybody is invited to contribute. It does not mean, in turn, that users present a rough idea and the devs do the rest. :wink: I always like see this a a "Do-ocracy": If you want something to change, care for it yourself. We have given you hints how to improve your pull request, so if you care enough about par times for Chex Quest (which you apparently do, else you wouldn't have done the first step by filing this request), please follow our advice.

Any change you commit to your own "Acts19quiz_chex_partimes" branch will modify this pull request once pushed to the repository on Github. Good luck!

File: src/doom/g_game.c
Comment (fabiangreffrath) [inline src/doom/g_game.c:1309]: Sorry, but this isn't even valid C code. You cannot have an if-condition in an initialization.

**Comment by Acts19quiz** [comment]
I'm aware of this, I'm currently setting up MSYS64 on my end so I can tinker around without tripping the Travis builder every time I want to tinker around.

Per other stuff, it's by no means at all that I want other people to do "the rest", dump a load off, etc. There appeared to be a minor disagreement on how to handle the array, so as much as anything else, apprehension stems from wanting you (all) to resolve things the way you want, and wanting to respect the way you want the code handled internally.

(also, I have amateur experience fiddling with id Tech 1 in C++, so while basic C logic is within my grasp, as primarily an armchair player like any other, if I come across something like say a complex rendering error in Crispy Doom, I'm outta luck and can only report, not pull--but that's another topic for another day)

**Comment by Acts19quiz** [comment]
I ran into errors compiling with MSYS2 (Mingw32), possibly related to [Crispy issue 425](https://github.com/fabiangreffrath/crispy-doom/issues/425), but the [archlinux archive](https://archive.archlinux.org/packages/s/sdl2/) only goes as far back as 2.0.9 for SDL2. That again, though, is another issue for another time. This time my commit is ready for review.

File: src/doom/g_game.c
Comment (fabiangreffrath) [inline src/doom/g_game.c:1324]: I'd go for `chexpars[6]`, remove the `{0}` and use `chexpars[gamemap]` later. You already checked that `gameepisode == 1` just a few lines earlier.

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]
A lot better already, thanks!

**Review by fabiangreffrath — APPROVED** [review]
LGTM

**Comment by fabiangreffrath** [comment]
Thank you!

**Comment by Acts19quiz** [comment]
No problem. The idea behind doing this here is that it will eventually filter its way into child ports (e.g., Crispy), and hopefully the addressed issue #1355 will be checked by community input on the next official release just to double check that it will work in normal gameplay as it looks like the fix should.

**Comment by fabiangreffrath** [comment]
Just merged the commit into crispy, thanks again!

**Comment by turol** [comment]
@Acts19quiz
As a followup can you make those arrays (all of them) `static const`?

**Comment by Acts19quiz** [comment]
I think changing the way the existing code works and numbers are stored is beyond the scope of this PR. As you mentioned, a new PR is probably in order for that (the one note I'll provide, possibly relevant, is that in testing Crispy Doom acknowledged BEX-extension par time additions, but not Chocolate Doom, though that appears to be a general [and intentional] lack of BEX support in Choco as opposed to anything BEX-specific).

**Pull request merged**
